### PR TITLE
Added images and links for seaborn, holoviews, and ggplot

### PR DESCRIPTION
**Pull request by jbednar**
As suggested in #6088, partially backported changes from #6088 to the v1.5.1-doc branch so that they will go public prior to the v2.0 release.

### CODE REVIEW

File: doc/mpl_toolkits/index.rst [hunk 1]
```diff
@@ -172,6 +172,26 @@ level interface for drawing statistical graphics with matplotlib. It
...
+.. image:: /_static/seaborn.png
+    :height: 157px
+
+.. _toolkit_holoviews:
+
+holoviews
+=========
+(*Not distributed with matplotlib*)
+
+`holoviews <http://holoviews.org>`_ makes it easier to visualize data
```
Comment (QuLogic) [inline doc/mpl_toolkits/index.rst]: I seem to recall @tacaswell saying something about these links requiring two `_` at the end.

Comment (tacaswell) [inline doc/mpl_toolkits/index.rst]: A single `_` is the simplest link. but if you tried to use the word 'holoviews' for a different link elsewhere on the same page sphinx would complain as it was now confused where 'holoviews' should point.  Using `__` makes it an 'anonymous' link so you can re-use the same text with a different target elsewhere on the page. (as explained by @jenshnielsen )

**Comment by tacaswell** [comment]
I tried to do the merge cascade, but v2.x does not cleanly merge into master.  I will come back to this tomorrow.

**Comment by tacaswell** [comment]
@jbednar Took me almost 2 months :sheep:, but I finally rebuilt the docs.  This version should be live now.

**Comment by jbednar** [comment]
Looks great, thanks!

**Pull request merged**